src/envoy/http/mixer/filter.cc

```diff
@@ -164,26 +165,70 @@ FilterTrailersStatus Filter::decodeTrailers(HeaderMap& trailers) {
   return FilterTrailersStatus::Continue;
 }
 
+FilterHeadersStatus Filter::encodeHeaders(HeaderMap& headers, bool) {
+  ENVOY_LOG(debug, "Called Mixer::Filter : {} {}", __func__, state_);
+  if (state_ == Calling) {
+    return FilterHeadersStatus::StopIteration;
```

Comment:
cc @PiotrSikora : I have a question here about the filter state machine. What is the order between encoding and decoding of headers? It seems that header encoding never kicks in prior to check completing. Calling continueEncoding on check complete leads to a segfault (assert failure). So it's not clear to me how to get out of `StopIteration` state here...

Reply:
This can never happen. So if you are in this state it is certainly a bug, so just do `ASSERT(state_ == Complete)`.

In the `continueEncoding` is a member of encoder_callbacks, which is not supposed to be called in decoder path (check complete). In decoder path, we stops at `decoderHeader` until check is complete, so the request won't go to upstream until check is done, hence the encoder stream would never be created at that moment.

Reply:
Done. Put an assert for complete/responded states.